packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -167,9 +175,11 @@ export function PrivateBlockToolbar( {
 				{ ! isMultiToolbar &&
 					isLargeViewport &&
 					isDefaultEditingMode && <BlockParentSelector /> }
-				{ isUsingBindings && canBindBlock( blockName ) && (
-					<BlockBindingsIndicator clientIds={ blockClientIds } />
-				) }
+				{ isUsingBindings &&
+					hasPatternOverrides &&
+					canBindBlock( blockName ) && (
+						<BlockBindingsIndicator clientIds={ blockClientIds } />
+					) }
```

Comment:
I think the `BlockBindingsIndicator` should be visible when using other types of bindings (e.g. meta) outside of the pattern block, so by adding `hasPatternOverrides` to the condition does it prevent that?

Reply:
Hmm, now I'm not sure what's expected for bindings. It should be one block icon, and I guess a purple one?

I'm not completely up to date on the block bindings UI changes, so would be good to get some confirmation.

cc @richtabor @SantosGuillamot

Reply:
Until now, the purple icon was just a connection icon.

![Screenshot 2024-06-05 at 16 53 08](https://github.com/WordPress/gutenberg/assets/34552881/b940a612-a3b1-4fbb-b040-4b9b99f3bdb5)

I can see a duplicated icon in `trunk` now, I guess since the linked pull request. I would rather show the connection icon than the duplicated one, but it'd be great to hear from @richtabor and @jasmussen .

Reply:
I prefer the after here, honestly. We've tried out the connection icon in this iteration in trunk for a bit, but it hasn't felt like it added clarity. I think there are other things we can do, but for now, I'd omit it.

Reply:
Okay, thanks for the feedback. Let me try to clarify to ensure we are aligned. For connections that are NOT pattern overrides, like post meta, there are three options so far:

**Option 1: The connection icon (the old implementation)**

![Screenshot 2024-06-05 at 16 53 08](https://github.com/WordPress/gutenberg/assets/34552881/b940a612-a3b1-4fbb-b040-4b9b99f3bdb5)

**Option 2: The duplicated icon (the existing implementation in trunk)**

![image](https://github.com/WordPress/gutenberg/assets/7753001/3f3421b1-7e17-484b-8f93-b0e1ffcf9b9b)

**Option 3: Removing the icon (this would be new)**

![image](https://github.com/WordPress/gutenberg/assets/7753001/bf739e80-9767-4a7c-a876-95c157fe1aad)

---
You refer to option 3, right?

Reply:
Yes, but I'm worried I'm missing nuance. @richtabor's original intent, as [outlined here](https://github.com/WordPress/gutenberg/pull/61560#issuecomment-2142427551), was that _the block icon was purple_. The main issue isn't that it's purple or not—I could go either way on that—it's that the icon is duplicated, it shouldn't be.

Reply:
I agree that the duplicated icon doesn't seem right.

I'm struggling to understand what is the desired outcome, though. Let me try to break it into three different use cases

**1. Pattern overrides when used in a post**

Right now in `trunk`, I can see the purple paragraph icon and when I click on it, it triggers the message. I assume this is the expected behavior, right?

![Screenshot 2024-06-06 at 10 44 05](https://github.com/WordPress/gutenberg/assets/34552881/66831ecf-9616-4636-b1d3-6e0ae81e2126)

**2. Editing a pattern override**

Right now in `trunk`, I can see the duplicated icon, which I guess it is what this pull request tries to solve.

![Screenshot 2024-06-06 at 10 47 01](https://github.com/WordPress/gutenberg/assets/34552881/019c5460-8ecf-49b5-b414-e7575974cb0c)

What is the desired behavior here? Just remove the purple icon or keep the purple icon and remove the "normal" one? If it is the second one, what should happen when users click on the icon? Should it still trigger the modal to transform the block?

**3. Using other bindings like post meta**

Right now in `trunk`, I can also see the duplicated icon.

I assume that whatever is decided for the second use case, "Editing a pattern override", should apply here as well?

Reply:
I'll let @richtabor respond as he drove some of the inital designs. For me, if the paragraph is connected, the block icon is purple because of that, but there is only ever a single block icon. If that shows a tooltip in the transforms dialog in one case, then it can show that same tooltip in both. If that replaces the transforms dropdown that is otherwise accessible in that space, then that's the main question to answer: should I not be able to transform blocks when they are connected?

Reply:
> should I not be able to transform blocks when they are connected?

In my opinion, we should definitely allow transforming blocks when they are connected. One of the benefits of using bindings is that users can transform a connected paragraph to a connected heading the same way they transform blocks right now, while keeping the connection.

Actually, it was even mentioned in the original proposal as one of the points: [link](https://github.com/WordPress/gutenberg/issues/51373).

Reply:
I would expect the purple block icon, if connected. It denotes the value is special, coming from somewhere else. 

Perhaps the popover help copy can go below transforms, if there are transforms.

Reply:
Overrides and connections should be essentially the same.

Reply:
Okay. Thanks for the feedback. So, if I understand you correctly, is this the desired behavior for an initial version until we gather more feedback? (Changing the texts I'm using maybe)

**1. Pattern overrides when used in a post**

![Screenshot 2024-06-06 at 10 44 05](https://github.com/WordPress/gutenberg/assets/34552881/66831ecf-9616-4636-b1d3-6e0ae81e2126)

**2. Editing a pattern override**

![Screenshot 2024-06-07 at 10 35 13](https://github.com/WordPress/gutenberg/assets/34552881/5d4ab99c-836d-410e-961d-fd31b3a58565)

**3. Using other bindings like post meta**

![Screenshot 2024-06-07 at 10 34 06](https://github.com/WordPress/gutenberg/assets/34552881/f7cf73e8-a8dc-4b37-a33e-4f654c21e926)

Reply:
That looks right to me, yes. The "This block is connected" message can be integrated into the transforms dropdown if need be.

Reply:
Perfect 🙂 @kevin940726 Let us know if it makes sense, if you plan to work on that, and if you need any help!

Reply:
I just pushed [c862489](https://github.com/WordPress/gutenberg/pull/62317/commits/c8624896c8513847662990ba3950aa2daae8d6a6). It now has the following UI:

Post meta connected block:
![image](https://github.com/WordPress/gutenberg/assets/7753001/df3a9098-20ac-463b-8abd-e41e53d74d6c)

Pattern overrides connected block in focus mode:
![image](https://github.com/WordPress/gutenberg/assets/7753001/8460d196-00a4-4062-b50b-bffe5fdda39d)

Pattern overrides connected block in post:
![image](https://github.com/WordPress/gutenberg/assets/7753001/8cefd9be-311e-4943-a073-8bf3c10be953)

LMK if I miss something!

Reply:
Regarding the UX, I believe it is much better now! Thanks for working on it. I'll take a look at the code asap.